jax/_src/xla_bridge.py

```diff
@@ -558,7 +590,10 @@ def backends() -> Dict[str, xla_client.Client]:
             raise RuntimeError(err_msg)
           else:
```

Comment:
I think this can be like:
```python
elif not fail_quietly:
  raise RuntimeError(err_msg)
else:
  _backends_errors[platform] = str(err)
  logger.info(err_msg)
```
(Like we don't need to set _backend_errors if we're raising, and then don't need the nested conditional below)

I also wonder if we should give some actionable guidance in the `not fail_quietly` case, maybe like:
```python
err_msg += " (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)"
```